src/main/java/loqor/ait/core/blockentities/LifeSupportBlockEntity.java

```diff
@@ -0,0 +1,83 @@
+package loqor.ait.core.blockentities;
+
+
+import org.jetbrains.annotations.Nullable;
+
+import net.minecraft.block.BlockState;
+import net.minecraft.entity.LivingEntity;
+import net.minecraft.entity.player.PlayerEntity;
+import net.minecraft.item.ItemStack;
+import net.minecraft.util.ActionResult;
+import net.minecraft.util.math.BlockPos;
+import net.minecraft.world.World;
+
+import loqor.ait.core.AITBlockEntityTypes;
+import loqor.ait.core.engine.SubSystem;
+import loqor.ait.core.engine.block.SubSystemBlockEntity;
+import loqor.ait.core.engine.impl.LifeSupportCircuit;
+import loqor.ait.core.engine.link.IFluidLink;
+import loqor.ait.core.engine.link.IFluidSource;
+import loqor.ait.core.engine.link.ITardisSource;
+import loqor.ait.core.tardis.Tardis;
+
+public class LifeSupportBlockEntity extends SubSystemBlockEntity implements ITardisSource {
```

Comment:
u dont want this to implement `ITardisSource` as it will get its fuel from the engine